src/generation/rag-chat.ts

```diff
@@ -46,28 +46,34 @@ export class RAGChat {
 
     const result = await this.model.run(messages.concat(message));
 
-    return result;
+    return { result, context };
   }
 
   stream(query: string, messages: OpenAIChatCompletionMessageInput[] = []) {
     const self = this;
+    const info: { context?: IVectorQueryResult[] } = {};
```

Comment:
Since this is now public, maybe we need a public type rather than `IVectorQueryResult` ?